### DISCUSSION TITLE
Logging additional  health data

### DISCUSSION THREAD

**Discussion by yoshimo** [Navid200/xDrip#98] [Ideas]
We have fitness trackers like a mi band  or a smart watch that collect the amount of steps you take, pulse, bloodpressure,sleep and various other values.

The mysugr diabetes diary has  additional fields to enter your weight, height, last hba1c (apart from the one it exprapolates based on your meals) , blood pressure, your last sport activity in steps and duration.
Splits the insulin amount in meal, correction and longterm, has a field for additional medizine you take, keton level.

Having dedicated data fields in the xdrip app and database would be great.

This would be great to better estimate your well being and insulin doses. If not via ai at least manually with brain.exe ;)

It could also connect to your watch and your fitness tracking devices to import these data if available and establish the foundation for the upcoming optical gcm apple bought into.

**Comment by Navid200** [comment]
This is my opinion only.  It's not an official response.

In my opinion, we should only add capabilities to xDrip that re absolutely necessary.
If something is only nice to have, in my opinion, we shouldn't add it.
My reason is that the xDrip code has already become too complicated.
As an example, I bring your attention to the following issue:
https://github.com/NightscoutFoundation/xDrip/issues/1266

That issue was opened more than 1.5 years ago.  It's not open because no one has looked at it.  It's still open because no one knows why it's happening!  The developers are not incompetent.  They are actually quite smart and competent.  
The problem is that so many different people have contributed to the code.  No one knows the entire code.  No one knows why ghost calibrations appear.  We only have speculations.

This will only get worse the more features we add to xDrip.

Compare this to what commercial companies do.  Look at Dexcom.  They had a G5 app.  When the next product came out, they released a new app for G6.  They didn't enhance the existing G5 app to also cover G6.  Why?

xDrip covers G4, G5, G6, MiaoMiao, Eversense, different Libre-related sources.  It covers Bluetooth glucose meters.  It shows heartbeat and steps.  It has a simulator.  It can follow Nightscout, Dex share, etc.

The G5 app from Dexcom didn't have any of these.  Yet, they just retired the app and came up with a new app to cover G6.  Very likely, when G7 comes out, they will release a new app for it.
We will have to add G7 to xDrip as well.

There is a price to pay for constantly adding more and more capabilities to a single app.

What do you want to have?  A great CGM that works reliably?  
Or something that does everything you dream of; but, none of them properly?

Of course I know what you want.  You want something that can do everything you can dream of reliably, right?  Unfortunately, that's not reality.  Everything has a price.

So, when someone says I want xDrip to store pictures I take from my food, after I read the request a few times to make sure it's not a joke, I say "why"?  Why can't you upload your pictures somewhere else and use a spreadsheet to associate the pictures with the time and dates you have had the food?  
I mean there will be no end to this.  Where do we draw the line?  What else are we going to add to xDrip?
What if someone asks for the telephone numbers of all his girlfriends to be stored in xDrip?  Should we really consider that?
When do we say no, that's enough?

**Reply by yoshimo** [reply]
The main goal of course is, cover the technical side and support the devices well and without issues.
Once that is covered we should also think about quality of life features other diabetes apps cover.

The open source aspect allows xdrip and others to connect to devices proprietary apps don't.
If companies don't talk to each other, you can't have nice things. If the manufacturer app is limited and requires the cloud, you are lost. Xdrip doesn't have these problems.
